### DISCUSSION TITLE
Building a CMake project

### DISCUSSION THREAD

**Discussion by NathanielAJohnson** [LazyVim/LazyVim#1702] [Q&A] [answered]
I am very new to Vim and **LazyVim** and come from a **vscode** and **clion** background.  I have successfully written a small embedded **C** project that also has a **CMake** project that configures the embedded board.  I have used nvim+LazyVim to write the code and it has worked splendidly.  I have been compiling and building using a separate terminal window which is fine but I was hoping to be able to build and compile from within nvim.  I feel like I am missing something obvious.  Does LazyVim come with a built in Task or CMake build system or do I need to add a plugin like [vim-cmake](https://github.com/cdelledonne/vim-cmake) or [asynctasks.vim](https://github.com/skywind3000/asynctasks.vim)?  If I need to add one is there a well supported plugin?  Is there an entirely different workflow that I am not aware of?  Advice would be appreciated.

I am working in Ubuntu if that makes a difference.

**Comment by amaanq** [comment, marked as answer]
You can enable the cmake extra which adds [cmake-tools](https://github.com/Civitasv/cmake-tools.nvim)

**Reply by NathanielAJohnson** [reply]
That see seems a good plug-in for cmake. Is there a more generic solution? I am thinking how LSP and DAP are built as providers for multiple tools. Specifically I use Make and Cargo as well.

**Reply by NathanielAJohnson** [reply]
I didn't see all the additional ___-extra addons. These do nicely.
